iml-api/src/graphql.rs

```diff
@@ -143,6 +144,93 @@ impl QueryRoot {
 
         Ok(xs)
     }
+    #[graphql(arguments(
+        limit(description = "paging limit, defaults to 20"),
+        offset(description = "Offset into items, defaults to 0"),
+        dir(description = "Sort direction, defaults to asc"),
+        args(description = "Snapshot listing arguments")
+    ))]
+    /// Fetch the list of snapshots
+    async fn snapshots(
+        context: &Context,
+        limit: Option<i32>,
+        offset: Option<i32>,
+        dir: Option<SortDir>,
+        args: List,
```

Comment:
I think we need to expand the list args into individual params in the interest of user-friendliness.

The downside of reusing the args / types from wire-types is we assume that the structs are one-size-fits-all, which is not always going to be the case.

Reply:
In case of snapshots, it seems fine. The structures are small and most fields are required. In any case, de-serializing will handle optional fields.

Reply:
How does this look in the `graphiql` client? I.E. how does the user input this?